rules/ruleset.go

```diff
@@ -50,6 +52,7 @@ var (
 	errRuleSetNotTombstoned = errors.New("ruleset is not tombstoned")
 	errRuleNotFound         = errors.New("rule not found")
 	errNoRuleSnapshots      = errors.New("rule has no snapshots")
+	errMissingOpInChanges   = merrors.NewInvalidInputError("changes must contain an op")
```

Comment:
I think a more proper error is `errUnknownOpType` since the op type is unknown. Also if you want to include the op type in the string, you can do `merrors.NewInvalidInputError(fmt.Sprintf("unknown op type %v", Op))` and move it to where this error is used.

Reply:
I was thinking about that. But it could be the case that the op is an empty string which would make this an awkward error message.